background/l10n.js

```diff
@@ -18,18 +18,24 @@ export default class BackgroundLocalizationProvider extends LocalizationProvider
     if (!locales.includes("en")) locales.push("en");
 
     localeLoop: for (const locale of locales) {
-      for (const addonId of addonIds) {
-        let resp;
-        let messages = {};
-        const url = `/addons-l10n/${locale}/${addonId}.json`;
-        try {
-          resp = await fetch(url);
-          messages = await resp.json();
-        } catch (_) {
-          if (addonId === "_general") continue localeLoop;
-          continue;
+      const cache = (await chrome.storage.session?.get("l10nCache"))?.l10nCache;
```

Comment:
This is a quite dangerous pattern. Reading the cache once for each locale, inside the loop? If we just checked Spanish and set the cache, and we're now checking English, how would the cache we just built be useful?
Splitting the cache by locale also sounds unnecessary.

I have the following idea instead.
- Each time `load()` is called (which might be multiple times per session, even though it's only called once for now) the value of `this.messages` should be stored in the cache, after the loop has finished.
- We should add a `loadFromCache()` method to l10n.js that takes an object, sets `this.messages` to it, calls `this._reconfigure()` and sets `this.loaded` appropriately.
- The cache should not be read by l10n.js. It should onlu be read by load-addon-manifests.js once on startup/wakeup. If the cache exists and the `addonIds` list matches the one that was used to build the cache (either because the code checked it, or we add a code comment that explains why it's guaranteed that if the cache exists then the addons.json file hasn't changed) then instead of calling load() it should call the new loadFromCache() method and pass the cached messages object to it.

Reply:
Also, to clarify, this for-loop exists to look for strings that may not be translated yet, so it fallbacks to English for those.

Reply:
I think I'm starting to understand this code...

Reply:
What would we do in the case where the user changes their display language while Scratch Addons is open? Where's the code that handles that?

Reply:
Browsers require a restart to change their UI language, so we don't need to consider that case.

Reply:
Does Scratch Addons even auto-update the language of its UI or does it require a reload?

Reply:
You should clarify your question, this is not simple. In some extreme cases, our strings could display in three different languages at the same time.
For example, if you set your Chrome language to French (the language used by Chrome for its UIs, which doesn't necessarily equal `navigator.languages[0]`) but you enable the `Show addon names and descriptions in English` toggle in More Settings, and your Scratch language is set to Japanese in the footer. In this case, the settings page will show some strings in French (those that go through `chrome.i18n.getMessage())` and some in English (those that were originally fetched by the background context). Strings in the Scratch website will attempt to match its Japanese surroundings.
In the popup, I believe the names of the tabs (Messaging, Games, etc.) will be in French while the strings used within the popup will be in English. Quite a mess. There's an issue about this somewhere.